service/openapi.yaml

```diff
@@ -0,0 +1,285 @@
+                format: date-time
+                example: '2024-02-07T12:00:00Z'
+        '500':
+          description: Internal server error
+  /ofrep/v1/flag/changes:
+    post:
+      description: OFREP polling endpoint to check if the flags have been modified
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/flags'
+      responses:
+        '200':
+          description: OFREP successful flag changes response
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/flagChange'
+        '400':
+          description: OFREP error response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/flagChangeFailure'
+        '401':
+          description: Unauthorized - You need credentials to access the API
+        '403':
+          description: Forbidden - You are not authorized to access the API
+        '429':
+          description: Rate limit reached on the Flag Management System
+          headers:
+            Retry-Later:
+              description: Indicates when to retry the request
+              schema:
+                type: string
+                format: date-time
+                example: '2024-02-07T12:00:00Z'
+        '500':
+          description: Internal server error
+components:
+  securitySchemes:
+    BearerAuth:
+      description: (optional) Bearer Authorization to your flag management system.
+      type: http
+      scheme: bearer
+    ApiKeyAuth:
+      description: (optional) API Key to your flag management system.
+      type: apiKey
+      in: header
+      name: X-API-Key
+  schemas:
+    bulkEvaluationRequest:
+      description: Evaluate multiple flags in one request
+      properties:
+        flags:
+          $ref: '#/components/schemas/flags'
+        context:
+          $ref: '#/components/schemas/context'
+    bulkEvaluationSuccess:
+      description: Response of the bulk evaluation
+      type: array
+      items:
+        properties:
+          key:
+            $ref: '#/components/schemas/key'
+          value:
+            type: object
+            description: Flag evaluation result
+          reason:
+            type: string
+            description: Reason matching OpenFeature reason
+          variant:
+            type: string
+            description: Variant of the evaluated flag value
+          metadata:
+            type: object
+            description: Arbitrary metadata supporting flag evaluation
+          errorCode:
+            type: string
+            description: OpenFeature compatible error code. See https://openfeature.dev/specification/types#error-code
+          errorDetails:
+            type: string
+            description: Optional error details description for logging or other needs
+          ETag:
```

Comment:
Shouldn't this be in the response headers, not in the payload body?

Reply:
We have put it in the body because we have 1 ETag per flag to avoid re-fetching/re-evaluating the full list if only 1 flag has changed.

I am open to alternative ideas here.

Reply:
The ETag should either be used in the header or we should come up with a different term. I would try to leverage ETags (as a header) if possible because it has broad compatibility and is built for caching use cases.

I don't think it's necessary to support change detection on each flag during a bulk evaluation. That would simplify the implementation and allow for the ETag to be on the response header.

Reply:
Will resolve as we followed up with the suggestions (and see [here](https://github.com/open-feature/protocol/pull/2#discussion_r1503180581) )